subsys/dfu/img_util/flash_img.c

```diff
@@ -234,6 +234,8 @@ int flash_img_init(struct flash_img_context *ctx)
 #ifdef CONFIG_IMG_ERASE_PROGRESSIVELY
 	ctx->off_last = -1;
 #endif
+#ifdef FLASH_AREA_IMAGE_SECONDARY
```

Comment:
why this change is here?

Reply:
I will remove this, it sort of depends on https://github.com/zephyrproject-rtos/zephyr/pull/23218 to make sense. 

In essence it allows the file to compile without having the mcuboot defines present.